libsoftwarecontainer/src/container.cpp

```diff
@@ -301,18 +306,26 @@ ReturnCode Container::executeInContainer(ContainerFunction function, pid_t *pid,
     options.uid = userID;
     options.gid = userID;
 
+    // List of vars to use when executing the function
     EnvironmentVariables actualVariables = variables;
 
-    // Add the variables set by the gateways
+    // Add the variables set by gateways, variables passed with the 'variables' argument
+    // will take precedence over previously set variables.
     for (auto &var : m_gatewayEnvironmentVariables) {
         if (variables.count(var.first) != 0) {
             if (m_gatewayEnvironmentVariables.at(var.first) != variables.at(var.first)) {
-                log_warning() << "Variable set by gateway overriding original variable value: "
-                              << var.first << ". values: " << var.second;
+                // Inform user that GW config will be overridden, it might be unintentionally done
+                log_info() << "Variable \""
+                           << var.first
+                           << "\" set by gateway will be overwritten with the value: \""
+                           << variables.at(var.first)
+                           << "\"";
             }
+            actualVariables[var.first] = variables.at(var.first);
```

Comment:
Is there any specific reason why use two different styles of accessing values from a map?